index.js

```diff
@@ -157,6 +151,14 @@ function defaultMethodNamesOrAssert(methodNames) {
 
 const isMetaEvent = eventName => eventName === listenerAdded || eventName === listenerRemoved;
 
+async function emitMetaEvent(emitter, eventName, eventData) {
+	if (isMetaEvent(eventName)) {
+		canEmitMetaEvents = true;
+		await emitter.emit(eventName, eventData);
```

Comment:
What should we do if this throws an error? `canEmitMetaEvents` won't be reverted to `false`...